clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -225,22 +226,27 @@ public static Properties consumerConfig(final String bootstrapServers, Class key
     }
 
     /**
-     *  uses default value of 30 seconds for timeout
+     *  uses default value of 15 seconds for timeout, and default message for failure
      */
     public static void waitForCondition(TestCondition testCondition) throws InterruptedException {
-        waitForCondition(testCondition, 30000);
+        waitForCondition(testCondition, 15000, "Expected condition not met by timeout");
```

Comment:
> Wouldn't it be better to also take a message in this method?

@ijuma 

Sure, I like that approach.  

I had originally wanted to use a type similar to a `Function` instead of a String:
```
List<KeyValue<K,V>> actualKeyValuesCollected = new ArrayList<>;

// use this to construct the assertion failure
ErrorMessage message = new ErrorMessage() {
  String apply() {
     int collected = actualKeyValuesCollected.size();
     return "Condition not met within timeout " + maxTimeMillis + "  expected  " 
+ expectedValuesCount+" but received " + collected";
     }
}
waitForCondition(testCondition, maxTimeMillis, message);


```
But I took this out this since Java 8 isn't required (no lambda expressions). 
So unless you feel this won't be too cumbersome, I'll go with the approach you suggested above.